src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -384,6 +397,45 @@ def delete_vcs_repository(cls, repo_name) -> None:
         except requests.exceptions.RequestException as err:
             raise RuntimeError(f'Failed to make vcs api call with error: {err}')
 
+    @classmethod
+    def create_cfs_source(cls, source_name):
+        """Creates a cfs source.
+
+        Args
+            source_name: name of the cfs source
+        Raises:
+            RuntimeError: if the source cannot be created
+        """
+        vcs_url = f"https://api-gw-service-nmn.local/vcs/cray/{cls.vcs_repo_name}.git"
+        command = (f"cray cfs v3 sources create "
+                   f"--name {source_name} "
+                   f"--clone-url {vcs_url} "
+                   f"--credentials-username {cls.vcs_username} "
+                   f"--credentials-password {cls.vcs_password}")
+
+        try:
+            subprocess.run(command, shell=True, check=True,
+                                    stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError as err:
+            raise RuntimeError(f'Failed to create cfs source with error: {err}')
+
+    @classmethod
+    def delete_cfs_source(cls, source_name):
```

Comment:
The `source_name` argument is not used since it's using the class attribute `cfs_source_name`.

```suggestion
    def delete_cfs_source(cls):
```